components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java

```diff
@@ -245,15 +247,15 @@ public void unregisterHookSupplier(final Supplier<StorageTxHook> hookSupplier) {
           tx.commit();
           return result;
         }
-        catch (OConcurrentModificationException e) {
-          lastConcurrentException = e;
+        catch (OConcurrentModificationException | ORecordDuplicatedException e) {
```

Comment:
Am a bit afraid about ORecordDuplicatedException... this implies that Operation must be Create-Or-Update aware, like I believe most are...